packages/core/src/browser/tree/tree-container.ts

```diff
@@ -7,31 +7,30 @@
 
 import { interfaces, Container } from 'inversify';
 import { TreeWidget, TreeProps, defaultTreeProps } from "./tree-widget";
-import { TreeModel, ITreeModel, TreeServices } from "./tree-model";
-import { Tree, ITree } from "./tree";
-import { ITreeSelectionService, TreeSelectionService } from "./tree-selection";
-import { ITreeExpansionService, TreeExpansionService } from "./tree-expansion";
+import { TreeModelImpl, TreeModel } from "./tree-model";
+import { TreeImpl, Tree } from "./tree";
+import { TreeSelectionService, TreeSelectionServiceImpl } from "./tree-selection";
+import { TreeExpansionService, TreeExpansionServiceImpl } from "./tree-expansion";
 import { TreeNavigationService } from './tree-navigation';
 import { TreeDecoratorService, NoopTreeDecoratorService } from './tree-decorator';
 
 export function createTreeContainer(parent: interfaces.Container): Container {
     const child = new Container({ defaultScope: 'Singleton' });
     child.parent = parent;
 
-    child.bind(Tree).toSelf();
-    child.bind(ITree).toDynamicValue(ctx => ctx.container.get(Tree));
+    child.bind(TreeImpl).toSelf();
+    child.bind(Tree).toDynamicValue(ctx => ctx.container.get(TreeImpl));
```

Comment:
fyi: `toService` for transitive

Reply:
https://github.com/theia-ide/theia/issues/1465